lib/diagnostic.rb

```diff
@@ -7,29 +7,34 @@ class Diagnostic
   # Assume each entry in the list implements a length method that returns
   # a Fixnum.
   def longest(list)
-    # Your code goes here
+    list.max { a. <=> length }
   end
 
   # Question 2
   # Write ruby code to return the sum of entries in the
   # list supplied (i.e. apply the `+` operator).
   # Assume each entry implements a `+` method.
   def sum(list)
-    # Your code goes here
+
   end
 
   # Question 3
   # Write ruby code that returns true if some of the entries
   # in the list of numbers are odd
   def some_odd(numbers)
-    # Your code goes here
+    odds = rng.select(&.odd?)
+
+  p odds
+
   end
 
   # Question 4
   # Write ruby code that returns true if every entry in the
   # list of numbers are even
   def every_even(numbers)
-    # Your code goes here
+    evens = rng.select(&.even?)
+
+  p evens
```

Comment:
Please read my comments for question 3, and look at the documentation for the `all?` method: http://ruby-doc.org/core-2.3.1/Enumerable.html#method-i-all-3F